### ISSUE TITLE
Undefined getdtablesize symbol

### ISSUE THREAD

**Issue by jperedadnr** [gluonhq/substrate#761]
Follow-up of https://github.com/gluonhq/client-maven-plugin/issues/236, when using JMX.

**Comment by jperedadnr** [comment]
This HelloWorld class can be used to reproduce the issue:

```
import com.sun.management.UnixOperatingSystemMXBean;

import java.lang.management.ManagementFactory;
import java.lang.management.OperatingSystemMXBean;

public class HelloWorld {

    private static final OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();


    public static void main(String[] args) {
        System.out.println("Hello World!! " + osMxBean + ", Unix: " + (osMxBean instanceof UnixOperatingSystemMXBean));
        UnixOperatingSystemMXBean unixOsBean = (UnixOperatingSystemMXBean) osMxBean;;
        unixOsBean.getOpenFileDescriptorCount();
        System.exit(0);
    }

}
```
It produces this runtime error on Android:

```
==================== RUN TASK ====================
[SUB] --------- beginning of main
[SUB] --------- beginning of system
[SUB] V/GraalActivity( 4413): onCreate start, using Android Logging v1
[SUB] V/GraalActivity( 4413): onCreate done
[SUB] V/GraalActivity( 4413): onStart
[SUB] V/GraalActivity( 4413): onStart done
[SUB] V/GraalActivity( 4413): onResume
[SUB] V/GraalActivity( 4413): onResume done
[SUB] V/GraalActivity( 4413): surfaceCreated for com.gluonhq.helloandroid.MainActivity@2baa7b5
[SUB] V/GraalActivity( 4413): loading substrate library
[SUB] --------- beginning of crash
[SUB] E/AndroidRuntime( 4413): FATAL EXCEPTION: main
[SUB] E/AndroidRuntime( 4413): Process: hello.helloworld, PID: 4413
[SUB] E/AndroidRuntime( 4413): java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "getdtablesize" referenced by "/data/app/hello.helloworld-Db4RLY2UlsTroZ5k5pqaLg==/lib/arm64/libsubstrate.so"...
[SUB] E/AndroidRuntime( 4413):  at java.lang.Runtime.loadLibrary0(Runtime.java:1071)
[SUB] E/AndroidRuntime( 4413):  at java.lang.Runtime.loadLibrary0(Runtime.java:1007)
[SUB] E/AndroidRuntime( 4413):  at java.lang.System.loadLibrary(System.java:1667)
[SUB] E/AndroidRuntime( 4413):  at com.gluonhq.helloandroid.MainActivity.surfaceCreated(MainActivity.java:104)
[SUB] E/AndroidRuntime( 4413):  at android.view.SurfaceView.updateSurface(SurfaceView.java:758)
```

**Comment by johanvos** [comment]
The reason for this is that GraalVM uses `getdtablesize` in `com.oracle.svm.core.posix.PosixSubstrateOperatingSystemMXBean.java` which is not supported (anymore) on Android.
At least on Android, we can use syscont(_SC_OPEN_MAX)

**Comment by johanvos** [comment]
While the fix for this is a PR in GraalVM (https://github.com/oracle/graal/issues/2812) we can work-around for now by defining getdtablesize in launcher.c

**Comment by johanvos** [comment]
Fixed by #766
